solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -611,8 +621,17 @@ contract RandomBeacon is Ownable {
         }
 
         if (relay.requestCount % groupCreationFrequency == 0) {
-            // TODO: Once implemented, invoke:
-            // createGroup(uint256(keccak256(entry)));
+            // Just in case, make sure the group creation can be triggered to
+            // avoid unexpected revert. There is a possibility entries are
+            // submitted very quickly and if the group creation frequency
+            // is small enough, group creations may overlap.
+            if (dkg.currentState() == DKG.State.IDLE) {
+                // slither-disable-next-line reentrancy-events
+                createGroup(uint256(keccak256(entry)));
```

Comment:
[RFC 19](https://github.com/keep-network/keep-core/blob/main/docs/rfc/rfc-19-random-beacon-v2.adoc) says:

>  Group creation start transaction should be embedded into relay request transaction and should lock a sortition pool. From this moment, no operator can enter or leave the pool. Once a new relay entry appears on the chain, all off-chain clients should perform group selection optimistically, using the new entry value and a view sortition pool function call. 

So the flow should be:
- Relay entry is requested. If `relay.requestCount % groupCreationFrequency == 0` and if `dkg.currentState() == DKG.State.IDLE`, we begin DKG by locking the pool.
- When relay entry is submitted, we request the creation of a new group with that relay entry. 
- DKG timeout starts counting from the moment relay entry is published.

Seed for group creation must not be known at the moment pool is locked to protect against the manipulation of sortition pool to get favorable group composition. With the current code, seed is not published to the chain but it is know - it sits in the mempool.

Reply:
Good point! I misunderstood the RFC. Will change.

Reply:
I have refactored the solution see: https://github.com/keep-network/keep-core/pull/2691/commits/f0a27d9d5da5eeca5344ca02271cbcbab02280d2. I'm still working on tests though. I think embedding pool lock/unlock logic within DKG state machine is the safest option. Keeping it separate seems error-prone because we can lock the pool forever or unlock it during group creation.

Reply:
Tests are ready as well: https://github.com/keep-network/keep-core/pull/2691/commits/efbe99ef5b55ea48841c74e94dc52202a7cbccc8